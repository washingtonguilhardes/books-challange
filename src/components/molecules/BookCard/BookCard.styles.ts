import { makeStyles } from '@material-ui/core/styles';

export const useBookCardStyles = makeStyles(() => ({
  root: {
    maxWidth: 345,
    width: '100%',
    height: '100%',
  },
  media: {
    height: 0,
    paddingTop: '56.25%', // 16:9
    backgroundPosition: 'center',
    backgroundSize: 'contain',
  },
}));
