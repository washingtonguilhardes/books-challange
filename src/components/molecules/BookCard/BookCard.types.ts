export interface BookCardItem {
  id: string;
  title: string;
  thumbnail: string;
}

export interface BookCardProps {
  book: BookCardItem;
}
