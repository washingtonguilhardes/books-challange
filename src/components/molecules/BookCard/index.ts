export * from './BookCard';
export * from './BookCard.types';
