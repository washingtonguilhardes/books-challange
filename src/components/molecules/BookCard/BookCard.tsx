import React from 'react';

import Card from '@material-ui/core/Card';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import FavoriteIcon from '@material-ui/icons/Favorite';

import { useBookCardStyles } from './BookCard.styles';
import { BookCardProps } from './BookCard.types';

export function BookCard(props: BookCardProps) {
  const { book, onClickFavorite, favorite } = props;
  const { thumbnail, title, id } = book;
  const classes = useBookCardStyles();

  return (
    <Card className={classes.root}>
      <CardMedia className={classes.media} image={thumbnail} title={title} />
      <CardActions disableSpacing>
        <IconButton
          aria-label="add to favorites"
          onClick={e => {
            e.preventDefault();
            e.stopPropagation();
            if (onClickFavorite) {
              console.log(id);
              onClickFavorite(id);
            }
          }}>
          <FavoriteIcon color={favorite ? 'primary' : 'disabled'} />
        </IconButton>
      </CardActions>
      <CardContent>
        <Typography variant="h5" color="textSecondary" component="p">
          {title.slice(0, 80)}
        </Typography>
      </CardContent>
    </Card>
  );
}
