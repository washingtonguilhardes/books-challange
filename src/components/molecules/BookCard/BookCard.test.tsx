import React from 'react';

import { render } from '@testing-library/react';

import { BookCard } from './BookCard';

describe('BookCard', () => {
  it('should match with snapshot', () => {
    const { container } = render(
      <BookCard book={{ id: '1', thumbnail: 'thumbnail', title: 'title' }} />
    );
    expect(container).toMatchSnapshot();
  });
});
