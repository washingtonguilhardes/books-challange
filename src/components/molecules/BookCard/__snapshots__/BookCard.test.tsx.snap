// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard should match with snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardMedia-root makeStyles-media-2"
      style="background-image: url(thumbnail);"
      title="title"
    />
    <div
      class="MuiCardActions-root"
    >
      <button
        aria-label="add to favorites"
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiCardContent-root"
    >
      <p
        class="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary"
      >
        title
      </p>
    </div>
  </div>
</div>
`;
